src/Workspaces/Core/Portable/Shared/TestHooks/AsynchronousOperationListener.cs

```diff
@@ -94,13 +102,24 @@ private void Decrement_NoLock(AsyncToken token)
             }
         }
 
-        public Task CreateWaitTask()
+        public Task CreateWaitTask(bool willBlockOnCompletion)
         {
+            if (willBlockOnCompletion && _counter > 0)
+            {
+                _blockedOnCompletionTokenSource.Cancel();
```

Comment:
I don't understand this? so, creating new wait task cancels all waiting. so, will bock on completion make it not block on delay? who use "will bock on completion : false" ?

Reply:
so, this is only case where it get cancelled?

Reply:
> so, creating new wait task cancels all waiting. so, will bock on completion make it not block on delay?

I don't understand the question. Can you rephrase it?

> who use "will bock on completion : false" ?

Removed this parameter

> so, this is only case where it get cancelled?

Yes

Reply:
This should be much clearer now.